### PR TITLE
Adding heroic coven

### PR DESCRIPTION
**Pull request by seanpeters86**

### CODE REVIEW

File: heroic/hc_coven_of_shivarra.simc [hunk 1]
```diff
@@ -1,5 +1,33 @@
 # movement
+# Whirling Saber
+raid_events+=/movement,cooldown=30,duration=1,first=10,last=200,players_only=1,player_chance=0.2
+raid_events+=/movement,cooldown=30,duration=1,first=400,players_only=1,player_chance=0.2
```
Comment (seanpeters86) [inline heroic/hc_coven_of_shivarra.simc:4]: There's a chance this doesn't spawn in your direction, but it typically does

File: heroic/hc_coven_of_shivarra.simc [hunk 2]
```diff
@@ -1,5 +1,33 @@
...
+# Storm of Darkness
+# could spawn one under you, so not totally 100%
+raid_events+=/movement,cooldown=60,duration=3,first=30,last=330,players_only=1,player_chance=0.8
```
Comment (seanpeters86) [inline heroic/hc_coven_of_shivarra.simc]: There is a slight chance that you will actually get storm to spawn under you so you don't need to move.

File: heroic/hc_coven_of_shivarra.simc [hunk 3]
```diff
@@ -1,5 +1,33 @@
...
+raid_events+=/adds,count=1,first=180,duration=20,cooldown=5000,max_distance=30
+raid_events+=/adds,count=1,first=180,duration=20,cooldown=5000,max_distance=30
+raid_events+=/adds,count=1,first=180,duration=30,cooldown=5000,max_distance=30
+# Flames of Khaz'goroth - spawn 8 assume 6 in range
+raid_events+=/adds,count=1,first=270,duration=15,cooldown=5000,max_distance=30
+raid_events+=/adds,count=1,first=270,duration=15,cooldown=5000,max_distance=30
+raid_events+=/adds,count=1,first=270,duration=20,cooldown=5000,max_distance=30
+raid_events+=/adds,count=1,first=270,duration=25,cooldown=5000,max_distance=30
+raid_events+=/adds,count=1,first=270,duration=25,cooldown=5000,max_distance=30
+raid_events+=/adds,count=1,first=270,duration=30,cooldown=5000,max_distance=30
```
Comment (seanpeters86) [inline heroic/hc_coven_of_shivarra.simc:29]: Decided not to add the Norganon adds, only useful for the sephuz proc as far as a sim is concerned

File: heroic/hc_coven_of_shivarra.simc (hunk 2 above)
Comment (nullKomplex) [reply heroic/hc_coven_of_shivarra.simc]: I would add some stddev for variance in duration, as you can be far/close to one.

**Comment by nullKomplex** [comment]
As long as we're assuming the bosses will never go invulnerable it looks solid to me. Also this isn't significant in this scenario, but I would like to note that putting the other boss at -18, -18 is ~27 yards apart not 18. Could also include blades but some attempts I forget they're a thing because they almost never hit.

**Review by nullKomplex — COMMENTED** [review]
Doesn't seem to work.

EDIT: Oh that didn't point to where I commented. The distance attribute doesn't seem to work.

EDIT2:
`  add_option( opt_float( "distance_min", distance_min ) );`

`  add_option( opt_float( "distance_max", distance_max ) );`

**Comment by seanpeters86** [comment]
Looks good now https://www.raidbots.com/simbot/report/2X4yWVmaxSxNA7J3rY2v9W

**Pull request merged**